pkg/gen/python-templates/CustomResource.py

```diff
@@ -15,37 +19,67 @@ class CustomResource(pulumi.CustomResource):
+            resource_name = __name__
+        if __opts__ is not None:
+            warnings.warn("explicit use of __opts__ is deprecated, use 'opts' instead", DeprecationWarning)
+            opts = __opts__
+        if not resource_name:
             raise TypeError('Missing resource name argument (for URN creation)')
-        if not isinstance(__name__, str):
+        if not isinstance(resource_name, str):
             raise TypeError('Expected resource name to be a string')
-        if __opts__ and not isinstance(__opts__, pulumi.ResourceOptions):
+        if opts and not isinstance(opts, pulumi.ResourceOptions):
             raise TypeError('Expected resource options to be a ResourceOptions instance')
 
         __props__ = dict()
 
-        if not api_version:
-            raise TypeError('Missing required property apiVersion')
         __props__['apiVersion'] = api_version
-        if not kind:
-            raise TypeError('Missing required property kind')
         __props__['kind'] = kind
-        if spec is None:
-            raise TypeError('Missing required property spec')
         __props__['spec'] = spec
         __props__['metadata'] = metadata
 
-        if __opts__ is None:
-            __opts__ = pulumi.ResourceOptions()
-        if __opts__.version is None:
-            __opts__.version = version.get_version()
+        opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(version=version.get_version()))
 
         super(CustomResource, self).__init__(
-            "kubernetes:%s:%s" % (api_version, kind),
-            __name__,
+            f"kubernetes:{api_version}:{kind}",
+            resource_name,
             __props__,
-            __opts__)
+            opts)
+
+    @staticmethod
+    def get(resource_name: str,
+            api_version: str,
+            kind: str,
+            id: pulumi.Input[str],
+            opts: Optional[ResourceOptions] = None):
+        """
+        :param str resource_name: _Unique_ name used to register this resource with Pulumi.
+        :param pulumi.Input[str] api_version: The API version of the apiExtensions.CustomResource we
+               wish to select, as specified by the CustomResourceDefinition that defines it on the
+               API server.
+        :param pulumi.Input[str] kind: The kind of the apiextensions.CustomResource we wish to select,
+               as specified by the CustomResourceDefinition that defines it on the API server.
+        :param pulumi.Input[str] id: An ID for the Kubernetes resource to retrieve.
+               Takes the form <namespace>/<name> or <name>.
+        :param Optional[ResourceOptions] opts: A bag of options that control this resource's behavior.
+        """
+
+        def _unwrap(id):
+            _opts = ResourceOptions(id=id) if opts is None else opts.merge(ResourceOptions(id=id))
+            return CustomResource(resource_name=resource_name, api_version=api_version, kind=kind, opts=_opts)
```

Comment:
we shouldn't actually need this.  I have opened https://github.com/pulumi/pulumi/pull/3116 to address this issue.  once that goes in we can remove this local function and write this as:

```py
opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(id=id))
return CustomResource(resource_name=resource_name, api_version=api_version, kind=kind, opts=opts)
```